src/Core/Framework/DataAbstractionLayer/Dbal/SchemaBuilder.php

```diff
@@ -193,25 +192,28 @@ public function buildSchemaOfDefinition(EntityDefinition $definition): Table
             );
         }
 
-        /** @var StorageAware[] $primaryKeys */
-        $primaryKeys = $definition->getPrimaryKeys()->filter(function (Field $field) {
-            return $field instanceof StorageAware;
-        })->getElements();
+        $primaryKeys = $definition->getPrimaryKeys()->fmap(static function (Field $field): ?string {
+            if ($field instanceof StorageAware) {
+                return $field->getStorageName();
+            }
 
-        $pk = PrimaryKeyConstraint::editor();
-        $pk->setUnquotedColumnNames(...array_values(array_map(function (StorageAware $field): string {
-            $name = $field->getStorageName();
-            \assert(!empty($name));
+            return null;
+        });
 
-            return $name;
-        }, $primaryKeys)));
-        $table->addPrimaryKeyConstraint($pk->create());
+        if ($primaryKeys) {
```

Comment:
if there are no primary keys, the `setUnquotedColumnNames` will get a failure, because no value is given